wrapanapi/systems/openstack.py

```diff
@@ -713,8 +775,27 @@ def remove_tenant(self, tenant_name):
         tid = self._get_tenant(name=tenant_name)
         self.tenant_api.delete(tid)
 
-    def create_vm(self):
-        raise NotImplementedError('create_vm not implemented.')
+    def create_vm(self, wait_for_state=VmState.RUNNING, wait_for_state_timeout=360,
+                  wait_for_state_delay=15, assign_floating_ip=False, floating_ip_pool=None,
+                  assign_floating_ip_timeout=120, **kwargs):
+        vm = self.connection.create_server(**kwargs)
```

Comment:
I see you're taking advantage of the connection object to call create_server, and passing it all the kwargs. Any reason you would not get an `OpenstackImage` object, and then use its `deploy` method to deploy the instance from that image?

Reply:
Thanks for the suggestion and Ok if this is the standard way to provision a VM, made some tests and seems also working.
```python
In[28]: image = os_system.get_template(name="rhel-7.6-server-x86_64-released")
In[29]: ins = image.deploy(
   ...:     "iqe-satellite-test-dlezz-124",
   ...:     flavor_name="m1.small",
   ...:     key_name=<hidden>,
   ...:     network_name=<hidden>
   ...: )
Unmapped VM state 'BUILD' received from system, mapped to 'VmState.UNKNOWN'
Unmapped VM state 'BUILD' received from system, mapped to 'VmState.UNKNOWN'
Unmapped VM state 'BUILD' received from system, mapped to 'VmState.UNKNOWN'
In[30]: ins.ip
Out[30]: '20.15.145.135'
# IP <modified>
In[31]: ins.name
Out[31]: 'iqe-satellite-test-dlezz-124'
In[32]: ins.exists
Out[32]: True
In[33]: ins.state
Out[33]: 'VmState.RUNNING'
In[34]: ins.uuid
Out[34]: '4942117f-a1fe-4757-bb85-b22ee2aea293'
In[35]: ins.cleanup()
Out[35]: True
In[36]: ins.exists
Out[36]: False
```
will update the code than.